arch/arm64/core/prep_c.c

```diff
@@ -45,6 +45,9 @@ static inline void z_arm64_bss_zero(void)
  */
 void z_arm64_prep_c(void)
 {
+	/* Initialize tpidrro_el0 with our struct _cpu instance address */
+	__asm__ volatile ("msr tpidrro_el0, %0" : : "r" (&_kernel.cpus[0]));
```

Comment:
Instead of using `__asm__` can you declare a reg helper with `MAKE_REG_HELPER(tpidrro_el0)` in `include/arch/arm64/lib_helpers.h` and use `{read,write}_tpidrro_el0()`?